source/Patches/RainbowMod/PlayerTabPatch.cs

```diff
@@ -1,39 +1,63 @@
-﻿using System;
-using HarmonyLib;
+﻿using HarmonyLib;
+using Reactor.Extensions;
+using System;
 using UnityEngine;
 using UnityEngine.Events;
 using Object = UnityEngine.Object;
 
 namespace TownOfUs.RainbowMod
 {
-    [HarmonyPatch(typeof(PlayerTab), nameof(PlayerTab.OnEnable))]
+    [HarmonyPatch(typeof(PlayerTab))]
     public class PlayerTabPatch
     {
-        public static bool Prefix(PlayerTab __instance)
+        [HarmonyPatch(nameof(PlayerTab.OnEnable))]
+        [HarmonyPostfix]
+        public static void OnEnablePatch(PlayerTab __instance)
         {
-            PlayerControl.SetPlayerMaterialColors(PlayerControl.LocalPlayer.Data.ColorId, __instance.DemoImage);
-            __instance.HatImage.SetHat(SaveManager.LastHat, PlayerControl.LocalPlayer.Data.ColorId);
-            PlayerControl.SetSkinImage(SaveManager.LastSkin, __instance.SkinImage);
-            PlayerControl.SetPetImage(SaveManager.LastPet, PlayerControl.LocalPlayer.Data.ColorId, __instance.PetImage);
+            foreach (ColorChip instanceColorChip in __instance.ColorChips)
+                instanceColorChip.gameObject.Destroy();
+
+            __instance.ColorChips.Clear();
+
             var colors = Palette.PlayerColors;
             var num = colors.Length / 4f;
-            for (int i = 0;i < colors.Length;i++)
+
+            for (int i = 0; i < colors.Length; i++)
             {
-                var x = __instance.XRange.Lerp((i % 4) / 4f) + 0.25f;
-                var y = __instance.YStart - (i / 4) * 0.55f;
+                var x = __instance.XRange.Lerp((i % 4) / 4f) + 0.26f;
+                var y = (__instance.YStart - 0.16f) - (i / 4) * 0.5f;
                 var colorChip = Object.Instantiate(__instance.ColorTabPrefab, __instance.ColorTabArea, true);
                 colorChip.transform.localScale *= 0.8f;
                 colorChip.transform.localPosition = new Vector3(x, y, -1f);
                 var colorId = (byte)i;
-                colorChip.Button.OnClick.AddListener((Action) (() =>
+
+                colorChip.Button.OnClick.AddListener((Action)(() =>
                 {
                     __instance.SelectColor(colorId);
-                    if (colorId <= 17) SaveManager.BodyColor = colorId;
+                    SaveManager.BodyColor = colorId <= 17 ? colorId : (byte)0;
```

Comment:
```suggestion
                    SaveManager.BodyColor = colorId <= 17 ? colorId : 0;
```
don't need to cast here if im understanding correctly

as a sidenote it might be nice to give players a way to save their custom color (eg create a custom save file/folder for ToU settings)

Reply:
> save their custom color\

Do you know where to set the saved color? I tried patching `PlayerControl.Start` with Postfix but it doesn't seems to work